bpf/init.sh

```diff
@@ -445,8 +447,16 @@ if [ "${TUNNEL_MODE}" != "<nil>" ]; then
 	if [ "$NODE_PORT" = "true" ]; then
 		COPTS="${COPTS} -DDISABLE_LOOPBACK_LB"
 	fi
-	bpf_load $ENCAP_DEV "$COPTS" "ingress" bpf_overlay.c bpf_overlay.o from-overlay ${CALLS_MAP}
-	bpf_load $ENCAP_DEV "$COPTS" "egress" bpf_overlay.c bpf_overlay.o to-overlay ${CALLS_MAP}
+
+	if ! bpf_load "$ENCAP_DEV" "$COPTS" ingress bpf_overlay.c bpf_overlay.o from-overlay "${CALLS_MAP}"; then
+		cilium bpf migrate-maps -e bpf_overlay.o -r 1
+	fi
+
+	if ! bpf_load "$ENCAP_DEV" "$COPTS" egress bpf_overlay.c bpf_overlay.o to-overlay "${CALLS_MAP}"; then
+		cilium bpf migrate-maps -e bpf_overlay.o -r 1
+	fi
```

Comment:
I think we are missing error cases here. If the load fails on ingress or egress, we want to exit the script right after running `cilium bpf migrate-maps`.

There's the same issue with all similar `if ! bpf_load` changes.

Reply:
Ah, unless `cilium bpf migrate-maps` return a non-zero code? :eyes:

Reply:
Ah yes, thanks!